modules/tensor_mechanics/include/materials/CalculateStrainLagrangianKernel.h

```diff
@@ -0,0 +1,106 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+#include "DerivativeMaterialInterface.h"
+#include "RankFourTensor.h"
+#include "RankTwoTensor.h"
+
+/// Calculate strains to use the MOOSE materials with the Lagrangian kernels
+//    This class wraps and alters strain information coming from the
+//    MOOSE material system to make it compatible with the
+//    UpdatedLagrangianStressDivergence and TotalLagrangianStressDivergence
+//    kernels
+//
+//    It has two basic jobs
+//    1) Calculate the deformation gradient at time steps n+1 and n
+//       (the MOOSE material system doesn't bother for the SmallStrain case)
+//    2) Calculate the kinematic quantities needed by the kernels:
+//      a) The incremental deformation gradient
+//      b) The inverse deformation gradient
+//      c) The determinant of the current deformation gradient
+//
+//    This object cooperates with the homogenization constraint system by
+//    adding in the scalar field representing the macroscale displacement
+//    gradient before calculating strains.
+//
+class CalculateStrainLagrangianKernel : public DerivativeMaterialInterface<Material>
```

Comment:
I'd not call a material "...Kernel". How about just "CalculateLagrangianStrain"?